### ISSUE TITLE
[FR] Add jason and base64 encoding API

### ISSUE THREAD

**Issue by sprunk** [beyond-all-reason/spring#877]
Seems to be a common and game-agnostic desire.

 * `Spring.EncodeJSON(table)` -> string
 * `Spring.DecodeJSON(string)` -> table
 * `Spring.EncodeBase64(string)` -> string
 * `Spring.DecodeBase64(string)` -> string

**Comment by Beherith** [comment]
Indeed, these would be sensible to have on the C++ side anyway. While the lua json ones arent terrible, I recall I had to modify the base64 one to not make so much garbage with repeated string concatenations.
